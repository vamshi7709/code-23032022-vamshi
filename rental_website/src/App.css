@import url("https://fonts.googleapis.com/css2?family=Dancing+Script&family=Exo+2&family=Ubuntu:wght@300;400&display=swap");


* {
  transition: ease-in;
}
.main {
  background: url("./assets/images/pexels-vecislavas-popa-1571460\ 1.svg");
  position: absolute;
  width: 1152px;
  height: 700px;
  left: 69px;
  top: 0;
  border-radius: 25px;
}

.span-arrow {
  position: absolute;
  width: 19px;
  left: 383px;
  top: 16px;
}
.logo {
  position: absolute;
  width: 186px;
  height: 53px;
  left: 31px;
  top: 15px;
  cursor: pointer;
}
.ul {
  list-style-type: none;
  position: absolute;
  display: flex;
  top: 19px;
  left: 551px;
}
.li {
  width: 104px;
  height: 39px;
  font-family: "Exo 2";
  font-style: normal;
  font-weight: 500;
  font-size: 20px;
  line-height: 110%;
  display: flex;
  align-items: center;
  text-align: center;
  color: #ffffff;
  margin-left: 26px;
}

#call {
  border: 2px solid #ffffff;
  box-sizing: border-box;
  border-radius: 10px;
  padding-left: 31px;
  height: 39px;
}
.header {
  position: absolute;
  width: 795px;
  height: 271px;
  left: 178px;
  top: 15px;

  font-family: "Exo 2";
  font-style: normal;
  font-weight: 700;
  font-size: 100px;
  line-height: 110px;
  display: flex;
  align-items: center;
  text-align: center;
  color: #ffffff;
}
.sub-header {
  position: absolute;
  width: 462px;
  height: 68px;
  left: 345px;
  top: 337px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 700;
  font-size: 30px;
  line-height: 110%;
  display: flex;
  align-items: center;
  text-align: center;
  color: #ffffff;
}

.rectangle-1 {
  position: absolute;
  width: 210px;
  height: 70px;
  left: 119px;
  top: 542px;
  background: #ffffff;
  border-radius: 16px;
  cursor: pointer;
}

.rectangle-2 {
  position: absolute;
  width: 210px;
  height: 70px;
  left: 340px;
  top: 542px;
  background: #ffffff;
  border-radius: 16px;
  cursor: pointer;
}

.rectangle-3 {
  position: absolute;
  width: 210px;
  height: 70px;
  left: 560px;
  top: 542px;
  background: #ffffff;
  border-radius: 16px;
  cursor: pointer;
}

.rectangle-4 {
  position: absolute;
  width: 235px;
  height: 70px;
  left: 791px;
  top: 542px;
  cursor: pointer;
  background: linear-gradient(90deg, #ff594c 0%, #fac527 100%);
  border-radius: 16px;
}

.rectangle-textup {
  position: absolute;
  left: 15px;
  top: -9px;
  font-family: "Exo 2";
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 110%;
  display: flex;
  align-items: center;
  text-align: center;
  color: #6d6d6d;
}

.rectangle-textbottom {
  position: absolute;
  width: 185px;
  left: 15px;
  top: 14px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 22px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}

.rectangle-4text {
  position: absolute;
  top: -6px;
  left: 95px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 500;
  font-size: 26px;
  line-height: 110%;
  display: flex;
  align-items: center;
  text-align: center;
  color: #ffffff;
}

.rectangle3 {
  position: absolute;
  width: 940px;
  height: 100px;
  left: 101px;
  top: 527px;
  background: linear-gradient(
    94.43deg,
    rgba(249, 249, 249, 0.49) 34.95%,
    rgba(249, 249, 249, 0.175) 67.01%
  );
  backdrop-filter: blur(10px);
  border-radius: 30px;
  border: 1px solid white;
}

.city {
  position: absolute;
  width: 210px;
  height: 70px;
  left: 109px;
  top: 542px;
  background: #ffffff;
  border-radius: 16px;
  border: transparent;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 22px;
  line-height: 110%;
}

.dates {
  position: absolute;
  width: 210px;
  height: 70px;
  left: 329px;
  top: 542px;
  background: #ffffff;
  border-radius: 16px;
  border: transparent;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 22px;
  line-height: 110%;
}

.guests {
  position: absolute;
  width: 210px;
  height: 70px;
  left: 549px;
  top: 542px;
  background: #ffffff;
  border-radius: 16px;
  border: transparent;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 22px;
  line-height: 110%;
}
.search {
  position: absolute;
  width: 235px;
  height: 70px;
  left: 783px;
  top: 542px;
  background: linear-gradient(90deg, #ff594c 0%, #fac527 100%);
  border-radius: 16px;
  border: transparent;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 22px;
  line-height: 110%;
}
.vec {
  position: absolute;
  width: 1152px;
  height: 51px;
  left: 0px;
  top: 650px;
}

.downarrow {
  position: absolute;
  width: 35px;
  height: 24.5px;
  left: 558px;
  top: 658px;
  cursor: pointer;
}

/* Explore CSS */

.explore {
  position: relative;
  width: 1152px;
  height: 700px;
  top: 700px;
  left: 79px;
}

.explore-header {
  position: absolute;
  width: 234px;
  height: 70px;
  left: 69px;
  top: 79px;
  font-family: "Exo 2";
  font-style: normal;
  font-weight: 700;
  font-size: 64px;
  line-height: 110%;
  display: flex;
  align-items: center;
  text-align: center;
  color: #000000;
}

.explore-p {
  position: absolute;
  width: 300px;
  height: 40px;
  left: 69px;
  top: 220px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}
.grid1 {
  position: absolute;
  width: 500px;
  height: 300px;
  left: 65px;
  top: 270px;
  background: url("./assets/images/pexels-dmitry-zvolskiy-2082087\ 1.png");
  border-radius: 25px;
}
.explore-grids {
  position: absolute;
  top: 70px;
}
.grid2 {
  position: absolute;
  width: 500px;
  height: 300px;
  left: 595px;
  top: 270px;
  background: url("./assets//images/pexels-naim-benjelloun-2029670\ 1.png");
  border-radius: 25px;
}
.grid1-trans {
  position: absolute;
  width: 424px;
  height: 100px;
  left: 32px;
  top: 250px;
  background: linear-gradient(
    94.33deg,
    rgba(255, 255, 255, 0.6) 3.19%,
    rgba(255, 255, 255, 0.1) 99.26%
  );
  backdrop-filter: blur(10px);
  border-radius: 15px;
  border: 1px solid white;
  padding: 1px;
  box-shadow: 0px 1px;
}

.grid-text {
  position: absolute;
  width: 300px;
  height: 20px;
  top: 2px;
  left: 21px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}

.grd-btn {
  position: absolute;
  width: 80px;
  height: 30px;
  left: 21px;
  top: 55px;
  padding-left: 16px;
  background: linear-gradient(90deg, #ff594c 0%, #fac527 100%);
  border-radius: 10px;
  border: transparent;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}
.grd-btn1 {
  position: absolute;
  width: 80px;
  height: 30px;
  left: 135px;
  top: 55px;
  padding-left: 16px;
  background: linear-gradient(90deg, #ff594c 0%, #fac527 100%);
  border-radius: 10px;
  border: transparent;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}
.grd-btn2 {
  position: absolute;
  width: 100px;
  height: 50px;
  left: 297px;
  top: 36px;
  background: linear-gradient(90deg, #ff594c 0%, #fac527 100%);
  border-radius: 10px;
  border: transparent;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 110%;
  padding-left: 25px;
  display: flex;
  align-items: center;

  color: #000000;
}

/* About CSS */

.about {
  position: relative;
  width: 1152px;
  height: 700px;
  top: 725px;
  left: 79px;
}

.about-header {
  position: absolute;
  width: 234px;
  height: 70px;
  left: 69px;
  top: 79px;
}

.about-h1 {
  font-family: "Exo 2";
  font-style: normal;
  font-weight: 700;
  font-size: 64px;
  line-height: 110%;
  width: 274px;
  height: 70px;
  display: flex;
  align-items: center;
  color: #000000;
}
.about-p {
  position: absolute;
  top: 111px;
  width: 300px;
  height: 20px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}

.about-grids {
  position: absolute;
  top: -35px;
}
.about-grid1 {
  position: absolute;
  width: 400px;
  height: 266.99px;
  top: 307px;
  left: 65px;
  background: url("./assets//images/pexels-max-vakhtbovych-5998051\ 1.png");
  border-radius: 15px;
}

.rectangle12 {
  position: absolute;
  width: 350px;
  height: 100px;
  left: 27px;
  top: 219px;
  background: linear-gradient(
    94.33deg,
    rgba(255, 255, 255, 0.6) 3.19%,
    rgba(255, 255, 255, 0.1) 99.26%
  );
  backdrop-filter: blur(10px);
  border-radius: 15px;
  border: 1px solid white;
  padding: 1px;
  box-shadow: 0px 1px;
}

.rectangle13 {
  position: absolute;
  width: 350px;
  height: 100px;
  left: 27px;
  top: 285px;
  background: linear-gradient(
    94.33deg,
    rgba(255, 255, 255, 0.6) 3.19%,
    rgba(255, 255, 255, 0.1) 99.26%
  );
  backdrop-filter: blur(10px);
  border-radius: 15px;
  border: 1px solid white;
  padding: 1px;
  box-shadow: 0px 1px;
}

.rectangle14 {
  position: absolute;
  width: 350px;
  height: 100px;
  left: 27px;
  top: 461px;
  background: linear-gradient(
    94.33deg,
    rgba(255, 255, 255, 0.6) 3.19%,
    rgba(255, 255, 255, 0.1) 99.26%
  );
  backdrop-filter: blur(10px);
  border-radius: 15px;
  border: 1px solid white;
  padding: 1px;
  box-shadow: 0px 1px;
}

.rectangle15 {
  position: absolute;
  width: 350px;
  height: 100px;
  left: 27px;
  top: 443px;
  background: linear-gradient(
    94.33deg,
    rgba(255, 255, 255, 0.6) 3.19%,
    rgba(255, 255, 255, 0.1) 99.26%
  );
  backdrop-filter: blur(10px);
  border-radius: 15px;
  border: 1px solid white;
  padding: 1px;
  box-shadow: 0px 1px;
}

.rectangle12-header {
  width: 128px;
  height: 26px;
  padding-left: 27px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}

.rectangle12-p {
  width: 300px;
  height: 30px;
  padding-left: 27px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 300;
  font-size: 14px;
  line-height: 110%;
  display: flex;
  align-items: center;
  color: #000000;
}

.about-grid2 {
  position: absolute;
  width: 500px;
  height: 333.74px;
  left: 555px;
  top: 318px;
  background: url("./assets//images/pexels-max-vakhtbovych-6492403\ 1.png");
  border-radius: 15px;
}
.about-grid3 {
  position: absolute;
  width: 400px;
  height: 502.58px;
  left: 65px;
  top: 701px;
  background: url("./assets/images/pexels-houzlook-com-3356416\ 2.png");
  border-radius: 15px;
  background-repeat: round;
}

.about-grid4 {
  position: absolute;
  width: 500px;
  height: 500px;
  left: 555px;
  top: 772px;
  background: url("./assets//images/pexels-oleg-zaicev-4834891\ 1.png");
  border-radius: 15px;
  background-repeat: round;
}

.bottom-nav {
  position: relative;
  width: 1152px;
  height: 200px;
  top: 1356px;
  left: 79px;
  border-bottom: 1px solid black;
  background: #ffffff;
  border-radius: 0px 0px 30px 30px;
  box-shadow: #000000;
}

.bottom-ul {
  list-style-type: none;
  position: absolute;
  display: flex;
  top: 31px;
  left: 359px;
}
.bottom-li {
  width: 104px;
  height: 39px;
  font-family: "Exo 2";
  font-style: normal;
  font-weight: 500px;
  font-size: 20px;
  line-height: 110%;
  display: flex;
  align-items: center;
  text-align: center;
  margin-left: 41px;
  text-decoration: none;
  padding: 10px;
}

#bottom-call {
  border: 2px solid #070000;
  box-sizing: border-box;
  border-radius: 10px;
  padding-left: 46px;
  /* height: 56px;
  width: 131px; */
  position: absolute;
  width: 131px;
  height: 50px;
  left: 524px;
  top: 5px;
}

.micropy {
  position: absolute;
  left: -32px;
  top: 3px;
  width: 25px;
}

.copyright {
  position: absolute;
  top: 87px;
  width: 173px;
  left: 64px;
  height: 53px;
  font-family: "Ubuntu";
  font-style: normal;
  font-weight: 300;
  font-size: 13px;
  line-height: 132%;
  /* or 18px */
  display: flex;
  align-items: center;
  color: #000000;
}

a:link {
  text-decoration: none;
}
