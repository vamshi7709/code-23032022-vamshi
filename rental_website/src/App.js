import React from 'react'
import './App.css';
import Main from './components/main';
import Explore from './components/explore';
import About from './components/about';
import BottomNav from './components/bottomNav';

function App() {
  return (
    <div className="App">
      <Main />
      <Explore />
      <About />
      <BottomNav />
    </div>
  );
}

export default App;
