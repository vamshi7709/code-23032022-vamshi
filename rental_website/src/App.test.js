import { render, screen, fireEvent, getByTestId } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Main from './components/main';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/penthouses/i);
  expect(linkElement).toBeInTheDocument();

  // const exploreElement = screen.getByText(/Explore/i);

  // expect(exploreElement).toBeCalledTimes(2);

});

test('renders explore', () => {
  render(<App />);

  const exploreElement = screen.getAllByText(/Explore/i);

  expect(exploreElement.length).toEqual(3);

});



/* test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
 */
