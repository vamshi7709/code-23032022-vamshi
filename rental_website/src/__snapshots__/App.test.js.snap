// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="App"
>
  <div
    className="main"
  >
    <div
      className="navbar"
    >
      <div
        className="logo"
      >
        <img
          alt="logo"
          className="logo"
          src="Your Logo.svg"
        />
      </div>
      <ul
        className="ul"
      >
        <a
          href="/#"
        >
          <li
            className="li"
          >
            Explore
          </li>
        </a>
        <a
          href="/#"
        >
          <li
            className="li"
            id="aboutus"
          >
            About Us
          </li>
        </a>
        <a
          href="/#"
        >
          <li
            className="li"
          >
            Cities
             
            <span>
              <img
                alt="down"
                className="span-arrow"
                src="image 2.svg"
              />
            </span>
          </li>
        </a>
        <a
          href="/#"
        >
          <li
            className="li"
            id="call"
          >
            Call
          </li>
        </a>
      </ul>
      <div
        className="headers"
      >
        <p
          className="header"
        >
          Rethink your living & renting
        </p>
        <p
          className="sub-header"
        >
          Make your stay painless with us
        </p>
      </div>
      <div
        className="rectangle3"
      />
      <div
        className="rectangle-1"
      >
        <p
          className="rectangle-textup"
        >
          CITY
        </p>
        <p
          className="rectangle-textbottom"
        >
          Select your city
        </p>
      </div>
      <div
        className="rectangle-2"
      >
        <p
          className="rectangle-textup"
        >
          DATES
        </p>
        <p
          className="rectangle-textbottom"
        >
          Select your dates
        </p>
      </div>
      <div
        className="rectangle-3"
      >
        <p
          className="rectangle-textup"
        >
          GUESTS
        </p>
        <p
          className="rectangle-textbottom"
        >
          Add guests
        </p>
      </div>
      <div
        className="rectangle-4"
      >
        <p
          className="rectangle-4text"
        >
          <span>
            <img
              alt="search"
              className="micropy"
              src="search.svg"
            />
          </span>
          Search
        </p>
      </div>
      <img
        alt="vec"
        className="vec"
        src="Vector (1).png"
      />
      <img
        alt="arrow"
        className="downarrow"
        src="image 2.svg"
      />
    </div>
  </div>
  <div
    className="explore"
  >
    <div
      className="explore-text"
    >
      <p
        className="explore-header"
      >
        Explore
      </p>
      <p
        className="explore-p"
      >
        From one-guest rooms to penthouses with pools and gardens
      </p>
    </div>
    <div
      className="explore-grids"
    >
      <div
        className="grid1"
      >
        <div
          className="grid1-trans"
        >
          <p
            className="grid-text"
          >
            Room with one king-size bed
          </p>
          <button
            className="grd-btn"
          >
            35$
          </button>
          <button
            className="grd-btn1"
          >
            28м²
          </button>
          <button
            className="grd-btn2"
          >
            Book!
          </button>
        </div>
      </div>
      <div
        className="grid2"
      >
        <div
          className="grid1-trans"
        >
          <p
            className="grid-text"
          >
            Penthouse for 8 person
          </p>
          <button
            className="grd-btn"
          >
            2039$
          </button>
          <button
            className="grd-btn1"
          >
            438м²
          </button>
          <button
            className="grd-btn2"
          >
            Book!
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="about"
    id="aboutus"
  >
    <div
      className="about-header"
    >
      <h1
        className="about-h1"
      >
        About Us
      </h1>
      <p
        className="about-p"
      >
        Allow us to tell you a short story...
      </p>
    </div>
    <div
      className="about-grids"
    >
      <div
        className="about-grid1"
      >
        <div
          className="rectangle12"
        >
          <h1
            className="rectangle12-header"
          >
            Chapter I
          </h1>
          <p
            className="rectangle12-p"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididun
          </p>
        </div>
      </div>
      <div
        className="about-grid2"
      >
        <div
          className="rectangle13"
        >
          <h1
            className="rectangle12-header"
          >
            Chapter II
          </h1>
          <p
            className="rectangle12-p"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididun
          </p>
        </div>
      </div>
      <div
        className="about-grid3"
      >
        <div
          className="rectangle14"
        >
          <h1
            className="rectangle12-header"
          >
            Chapter III
          </h1>
          <p
            className="rectangle12-p"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididun
          </p>
        </div>
      </div>
      <div
        className="about-grid4"
      >
        <div
          className="rectangle15"
        >
          <h1
            className="rectangle12-header"
          >
            Chapter IV
          </h1>
          <p
            className="rectangle12-p"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididun
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="bottom-nav"
  >
    <div
      className="bottom-navbar"
    >
      <div
        className="logo"
      >
        <img
          alt="logo"
          className="logo"
          src="Your Logo2.svg"
        />
      </div>
      <ul
        className="bottom-ul"
      >
        <li
          className="bottom-li"
        >
          Explore
        </li>
        <li
          className="bottom-li"
        >
          About Us
        </li>
        <li
          className="bottom-li"
        >
          Cities 
        </li>
        <li
          className="bottom-li"
          id="bottom-call"
        >
          Call
        </li>
      </ul>
      <p
        className="copyright"
      >
        © 2021 Company Name All rights reserved
      </p>
    </div>
  </div>
</div>
`;
